support-frontend/app/controllers/DigitalSubscriptionController.scala

```diff
@@ -3,174 +3,84 @@ package controllers
-import views.html.helper.CSRF
-import views.html.subscriptionCheckout
-import com.gu.support.zuora.api.ReaderType.{Direct, Gift}
 
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.ExecutionContext
 
 class DigitalSubscriptionController(
   priceSummaryServiceProvider: PriceSummaryServiceProvider,
   landingCopyProvider: LandingCopyProvider,
   val assets: AssetsResolver,
   val actionRefiners: CustomActionBuilders,
-  identityService: IdentityService,
-  testUsers: TestUserService,
-  membersDataService: MembersDataService,
-  stripeConfigProvider: StripeConfigProvider,
-  payPalConfigProvider: PayPalConfigProvider,
   components: ControllerComponents,
   stringsConfig: StringsConfig,
   settingsProvider: AllSettingsProvider,
   val supportUrl: String,
-  fontLoaderBundle: Either[RefPath, StyleContent],
-  recaptchaConfigProvider: RecaptchaConfigProvider
-)(
-  implicit val ec: ExecutionContext
-) extends AbstractController(components) with GeoRedirect with CanonicalLinks with Circe with SettingsSurrogateKeySyntax {
+  fontLoaderBundle: Either[RefPath, StyleContent]
+)(implicit val ec: ExecutionContext) extends AbstractController(components) with GeoRedirect with CanonicalLinks with SettingsSurrogateKeySyntax {
 
   import actionRefiners._
 
   implicit val a: AssetsResolver = assets
 
+  def digitalGeoRedirect(orderIsAGift: Boolean = false): Action[AnyContent] = geoRedirect(
+    if (orderIsAGift) "subscribe/digital/gift" else "subscribe/digital"
+  )
+
   def digital(countryCode: String, orderIsAGift: Boolean): Action[AnyContent] = CachedAction() { implicit request =>
     implicit val settings: AllSettings = settingsProvider.getAllSettings()
-    val title = s"Support the Guardian | The Guardian Digital ${if (orderIsAGift) "Gift " else ""}Subscription"
-    val mainElement = EmptyDiv("digital-subscription-landing-page-" + countryCode)
-    val js = Left(RefPath("digitalSubscriptionLandingPage.js"))
-    val css = Left(RefPath("digitalSubscriptionLandingPage.css"))
-    val description = stringsConfig.digitalPackLandingDescription
-    val canonicalLink = Some(buildCanonicalDigitalSubscriptionLink("uk", orderIsAGift))
-    val queryPromos = request.queryString.get("promoCode").map(_.toList).getOrElse(Nil)
-    val hrefLangLinks = Map(
-      "en-us" -> buildCanonicalDigitalSubscriptionLink("us", orderIsAGift),
-      "en-gb" -> buildCanonicalDigitalSubscriptionLink("uk", orderIsAGift),
-      "en-au" -> buildCanonicalDigitalSubscriptionLink("au", orderIsAGift),
-      "en" -> buildCanonicalDigitalSubscriptionLink("int", orderIsAGift)
-    )
-    val shareImageUrl = Some("https://i.guim.co.uk/img/media/74422ad120c709448f433c34f5190e2465ffa65e/0_0_1200_1200/1200.png?width=1200&auto=format&fit=crop&quality=85&s=1407add4d016d15cc074b0f9de8f1433") // scalastyle:ignore
-    if (settings.switches.enableDigitalSubGifting.isOn || !orderIsAGift) {
+
+    if (!settings.switches.enableDigitalSubGifting.isOn && orderIsAGift) {
```

Comment:
this if statement logic was a bit upside down, so I flipped it for clarity.
i.e. `!(a || b) == (!a && !b)`
I guess we will get rid of the switch soon anyway?